### DISCUSSION TITLE
Duplicating records when the title is unique

### DISCUSSION THREAD

**Discussion by e93Ym0WnIO** [payloadcms/payload#2244] [Feature Requests & Ideas]
Hi, I would like to suggest a feature for Payload. 

Lately I've been working a lot with it, and I noticed something when duplicating records.

If I have as follows:
```
const Exercises: CollectionConfig = {
  slug: 'exercises',
  admin: {
    useAsTitle: 'title',
  },
  fields: [
    {
      name: 'title',
      type: 'text',
      unique: true,
    },
  ],
}
```
And I try to duplicate a record, using the "Duplicate" button that is in the top-right of the admin panel, I get an error: `The following field is invalid: title`, but at the same time I get a `successfully duplicated` message.

I think it would be nice if, after clicking "Duplicate", payload would open an interface with the same fields of the record I am trying to duplicate, but without actually saving it.

Opening an identical record would allow me to edit only the fields required to be unique, and then save it manually. This would solve the issue with duplicating records that contain unique fields.

Have a great day

**Comment by mattddean** [comment]
We've encountered the same thing and agree that it would make sense if Duplicate did not try to immediately save the duplicated document

**Comment by DanRibbens** [comment]
I agree that the UX for duplicate could be improved. 

We did add a frontend hook for `beforeDuplicate` that can help solve this in a programmatic way. See docs: https://payloadcms.com/docs/configuration/collections#admin-hooks

Does this solve the issue for you @github-luciano and @mattddean?

**Reply by mattddean** [reply]
Yup, this will work for me, thanks @DanRibbens!

**Reply by e93Ym0WnIO** [reply]
Works wonders, thanks!
